File: library/src/main/java/com/sothree/slidinguppanel/canvassaveproxy/CanvasSaveProxyFactory.java
package com.sothree.slidinguppanel.canvassaveproxy;

import android.graphics.Canvas;
import android.os.Build;

public class CanvasSaveProxyFactory {

    @SuppressWarnings("deprecation")
    public CanvasSaveProxy create(final Canvas canvas) {

        if (Build.VERSION.SDK_INT >= 28) {
            return new AndroidPCanvasSaveProxy(canvas);
        } else {
            return new LegacyCanvasSaveProxy(canvas);
        }
    }
}

File: library/src/main/kotlin/com/sothree/slidinguppanel/canvasSaveProxy/CanvasSaveProxyFactory.kt
package com.sothree.slidinguppanel.canvasSaveProxy

import android.graphics.Canvas
import android.os.Build

class CanvasSaveProxyFactory {
    fun create(canvas: Canvas): CanvasSaveProxy {
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
            AndroidPCanvasSaveProxy(canvas)
        } else {
            LegacyCanvasSaveProxy(canvas)
        }
    }
}
